### DISCUSSION TITLE
Create a ticket to to eclipse.org to integrate the ecj patch

### DISCUSSION THREAD

**Discussion by brianantony456** [salesforce/bazel-jdt-java-toolchain#27] [General]
We use this bazel-jdt-java-tooclhain and replace the ecj.jar with our latest internal jar. Howver with the change below we get a compile error as expected.
```java
java.lang.IllegalAccessError: class com.google.devtools.build.buildjar.javac.BlazeEcjMain$BlazeEclipseBatchCompiler tried to access private method 'void org.eclipse.jdt.internal.compiler.batch.Main.setSeverity(java.lang.String, int, boolean)' 
```
Do you think it makes sense to integrate this change to the global ecj.jar ? If so could you be able to create a ticket in eclipse.org to integrate this change and avoid the patch ?

https://github.com/salesforce/bazel-jdt-java-toolchain/commit/20c1eaa4e19362dc94723d2b7366f99b9ad3ffac

**Comment by guw** [comment]
I think it will be difficult to get this accepted by the Eclipse JDT folks. It's needs a lot polishing. It might be easier to get something accepted that is less specific but introduces a more general listener/plug-in concept into ECJ. But I don't have the time currently to work on this.

Would you be interested in adding a PR updating the ECJ version to your needs?

**Reply by brianantony456** [reply]
Unfortunately, I am not a java deveoper and barely understand the changes. I talked to a member of the eclipse team, they are willing to make the change themselves provided a ticket is created. 
 
Could you please then create a ticket, requesting the following:
- what change is required.
- why & where it is used.

Also could you please let me know once the ticket is ready so that I forward it.

**Comment by iloveeclipse** [comment]
> I think it will be difficult to get this accepted by the Eclipse JDT folks. 

I'm a one of ecj folks: please create ticket explaining your needs and PR containing current changes. We can discuss on the PR whether it can be merged "as is" or polish it till it can be merged.

> It's needs a lot polishing. It might be easier to get something accepted that is less specific but introduces a more general listener/plug-in concept into ECJ. But I don't have the time currently to work on this.

Me and Simeon could take over with your PR (once you push it :-) ), but we need a ticket with explanation what is actually needed and why, as we are far from understanding all of bazel specialities.

**Reply by guw** [reply]
Alright. Let's move the discussion then. I described the requirement here.
https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2529

**Reply by iloveeclipse** [reply]
Patch was just merged upstream, see https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2577
